src/createAnimatedComponent.js

```diff
@@ -109,8 +122,13 @@ export default function createAnimatedComponent(Component) {
         if (prop instanceof AnimatedEvent && !attached.has(prop.__nodeID)) {
           // not yet attached
           prop.attachEvent(node, key);
+          nativeUpdate[key] = true;
         }
       }
+
+      if (Object.keys(nativeUpdate).length > 0) {
+        this.setNativeProps(nativeUpdate);
```

Comment:
Are we sure that `setNativeProps` is available here? Especially regarding [comment below](https://github.com/software-mansion/react-native-reanimated/pull/528/files#diff-3be87c4c6b24297654a7e0b2990508f2R134-R138).

Reply:
I had an app run with this. Why wouldn't it? If I recall correctly this is how event registration is done.

Reply:
I will test that because `this` here should be a composite component which theoretically doesn't provide this method 😄

Reply:
It's exposed [here](https://github.com/software-mansion/react-native-reanimated/blob/76b0dd15e367b9dd09f0bc2d8a3a88d5d076bf18/src/createAnimatedComponent.js#L45)

Reply:
Ah, wrong `this`.

I meant that `this._component.setNativeProps(props);` may fail.

Add definition of the wrapper:

```jsx
class TextWrapper extends Component {
  constructor(props) {
    super(props);
    const {text, ...rest} = props;
    this.text = text;
    this.rest = rest;
  }
  render() {
    return <Text {...this.rest}>{this.text}</Text>;
  }
}
const AnimTextWrapper = createAnimatedComponent(TextWrapper);
```

and replace one of the `Text`s by it:

```jsx
  <AnimTextWrapper
          onLayout={this._layoutEvent}
          style={[
            styles.box,
            {
              width: this._trans,
              height: this._trans,
            },
          ]}
          text={`
          sample text is getting bigger and bigger more and moar staph staph
          stophhh`}
        />
```

You can see that `_attachNativeEvents` will fail.

![image](https://user-images.githubusercontent.com/12465392/89875157-3011db80-dbbd-11ea-99ac-c45dc4d4a2d1.png)


Probably the best solution is to extract logic from `_animatedPropsCallback` to function and call it `_setNativePropsOrRerender` or sth and use it instead `_setNativeProps`. Works fine on this example, don't know if it can introduce some edge cases which will cause infinite rerenders, worth checking out.